src/Microsoft.Health.Fhir.TemplateManagement/Overlay/OverlayOperator.cs

```diff
@@ -21,61 +22,74 @@ namespace Microsoft.Health.Fhir.TemplateManagement.Overlay
 {
     public class OverlayOperator : IOverlayOperator
     {
-        public OCIFileLayer ExtractOCIFileLayer(OCIArtifactLayer artifactLayer)
+        public OciFileLayer Extract(ArtifactBlob artifactLayer)
         {
             EnsureArg.IsNotNull(artifactLayer, nameof(artifactLayer));
 
             if (artifactLayer.Content == null)
             {
-                return new OCIFileLayer();
+                return new OciFileLayer();
             }
 
             var artifacts = StreamUtility.DecompressTarGzStream(new MemoryStream(artifactLayer.Content));
             var metaBytes = artifacts.GetValueOrDefault(Constants.OverlayMetaJsonFile);
-            OverlayMetadata metaJson = null;
+
             if (metaBytes != null)
             {
                 try
                 {
-                    metaJson = JsonConvert.DeserializeObject<OverlayMetadata>(Encoding.UTF8.GetString(metaBytes));
+                    var metaJson = JsonConvert.DeserializeObject<OverlayMetadata>(Encoding.UTF8.GetString(metaBytes));
                 }
                 catch (Exception ex)
                 {
                     throw new OverlayException(TemplateManagementErrorCode.OverlayMetaJsonInvalid, $"{Constants.OverlayMetaJsonFile} file invalid.", ex);
                 }
             }
 
-            return new OCIFileLayer()
+            return new OciFileLayer()
             {
                 Content = artifactLayer.Content,
                 FileContent = artifacts,
                 FileName = artifactLayer.FileName,
-                SequenceNumber = metaJson?.SequenceNumber ?? -1,
                 Digest = artifactLayer.Digest,
                 Size = artifactLayer.Size,
             };
         }
 
-        public List<OCIFileLayer> ExtractOCIFileLayers(List<OCIArtifactLayer> artifactLayers)
+        public List<OciFileLayer> Extract(List<ArtifactBlob> artifactLayers)
         {
             EnsureArg.IsNotNull(artifactLayers, nameof(artifactLayers));
 
-            return artifactLayers.Select(ExtractOCIFileLayer).ToList();
+            return artifactLayers.Select(Extract).ToList();
         }
 
-        public List<OCIFileLayer> SortOCIFileLayersBySequenceNumber(List<OCIFileLayer> fileLayers)
+        public List<ArtifactBlob> Sort(List<ArtifactBlob> imageLayers, ManifestWrapper manifest)
         {
-            EnsureArg.IsNotNull(fileLayers, nameof(fileLayers));
+            var sortedLayers = new List<ArtifactBlob>();
+            ValidationUtility.ValidateManifest(manifest);
+            foreach (var layerInfo in manifest.Layers)
+            {
+                var currentLayers = imageLayers.Where(layer => layer.Digest == layerInfo.Digest).ToList();
```

Comment:
nit: it might be more straight-forward to use `FirstOrDefault()` instead of `ToList()` to get the layer and check whether `currentLayer` is null.